src/lib/io/sol-uart-impl-linux.c

```diff
@@ -172,72 +269,74 @@ sol_uart_open(const char *port_name, const struct sol_uart_config *config)
 SOL_API void
 sol_uart_close(struct sol_uart *uart)
 {
+    struct sol_blob *blob;
+    uint16_t i;
+
     SOL_NULL_CHECK(uart);
-    if (uart->async.rx_fd_handler)
-        sol_fd_del(uart->async.rx_fd_handler);
-    if (uart->async.tx_fd_handler)
-        sol_fd_del(uart->async.tx_fd_handler);
+
+    if (uart->rx.used)
+        uart->rx_cb((void *)uart->user_data, uart, &uart->rx);
+
+    if (uart->fd_handler)
+        sol_fd_del(uart->fd_handler);
+
+    if (uart->read_timeout)
+        sol_timeout_del(uart->read_timeout);
+
+    SOL_PTR_VECTOR_FOREACH_IDX (&uart->pending_blobs, blob, i)
+        sol_blob_unref(blob);
```

Comment:
I'd say we need to callback and report they were not dispatched, like ECANCELLED